Source/WebCore/rendering/TextAutoSizing.cpp

```diff
@@ -160,7 +160,7 @@ auto TextAutoSizingValue::adjustTextNodeSizes() -> StillHasNodes
             continue;
 
         auto newParentStyle = cloneRenderStyleWithState(parentStyle);
-        newParentStyle.setLineHeight(lineHeightLength.isNegative() ? Length(lineHeightLength) : Length(lineHeight, LengthType::Fixed));
+        newParentStyle.setLineHeight(lineHeightLength.isNormal() ? Length(lineHeightLength) : Length(lineHeight, LengthType::Fixed));
         newParentStyle.setSpecifiedLineHeight(Length { lineHeightLength });
```

Comment:
I think this line needs to change to:
```
newParentStyle.setSpecifiedLineHeight(lineHeightLength.isNormal() ? Length { LengthType::Normal } : Length { lineHeightLength });
```

Otherwise, it discards the "normal" type. The old code used to work, because the other checks just looked at whether the value was negative.

Reply:
That hopefully should fix all the iOS failures.

Reply:
Hmm you're right, let me take a deeper look